test/Node/NodeSpec.hs

```diff
@@ -0,0 +1,157 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Node.NodeSpec (spec) where
+
+import Algebra.Graph.AdjacencyMap qualified as AM
+import Data.Foldable (for_)
+import Data.Glob (unsafeGlobRel)
+import Data.Map qualified as Map
+import Graphing qualified
+import Path (Abs, Dir, Path, mkRelDir, mkRelFile, (</>))
+import Path.IO (getCurrentDir)
+import Strategy.Node (NodeProject (NPMLock), discover, getDeps)
+import Strategy.Node.PackageJson (Manifest (..), PackageJson (PackageJson, packageDeps, packageDevDeps, packageLicense, packageLicenses, packageName, packagePeerDeps, packageVersion, packageWorkspaces), PkgJsonGraph (PkgJsonGraph, jsonGraph, jsonLookup), PkgJsonLicense (LicenseText), PkgJsonWorkspaces (PkgJsonWorkspaces, unWorkspaces))
+import Test.Effect (it', shouldBe')
+import Test.Hspec (Spec, describe, runIO)
+import Types (DependencyResults (DependencyResults, dependencyGraph, dependencyGraphBreadth, dependencyManifestFiles), DiscoveredProject (DiscoveredProject, projectBuildTargets, projectData, projectPath, projectType), DiscoveredProjectType (NpmProjectType), FoundTargets (ProjectWithoutTargets), GraphBreadth (Complete))
+
+spec :: Spec
+spec = do
+  currDir <- runIO getCurrentDir
+  pkgJsonWorkspaceSpec currDir
+  npmLockAnalysisSpec currDir
+
+discoveredWorkSpaceProj :: Path Abs Dir -> DiscoveredProject NodeProject
+discoveredWorkSpaceProj currDir =
+  DiscoveredProject
+    { projectType = NpmProjectType
+    , projectPath = currDir </> $(mkRelDir "test/Node/testdata/workspace-test/")
+    , projectBuildTargets = ProjectWithoutTargets
+    , projectData =
```

Comment:
[no action]: I don't love this massive literal data block, but I don't really have a good alternative.

Reply:
It's not my favorite and I can't think of an alternative either. I've reduced a bit of the vertical space it takes up, but that doesn't change much fundamentally.